WB-1049 - Add `id`, `type`, `value`, and `onChange` Props to TextField

RivasCVA opened the pull request:
## Summary:
* Added the Props `id`, `type`, `value`, and `onChange` to the TextField.
* Included unit tests for all of these new props (Coverage displays 100% for the TextField component after the new unit tests).
* Added more examples to Styleguidist and React Storybook.

Issue: WB-1049

## Test plan:
1. Open Styleguidist in the browser
2. Check the new examples under Form -> TextField
3. You can also view the new examples on React Storybook under Form -> Textfield

##### These are the new additions I added to Styleguidist and React Storybook

###### Number
<img width="391" alt="tf-number" src="https://user-images.githubusercontent.com/60367213/119705566-84b9d680-be1e-11eb-957d-f21c690464cb.png">

###### Password
<img width="390" alt="tf-password" src="https://user-images.githubusercontent.com/60367213/119705570-86839a00-be1e-11eb-84c9-c170aebbae87.png">

##### NOTE: Will also add `email` and `tel` examples when the `validation` prop is added.

jandrade reviewed (approved):
LGTM! Just a minor suggestions for the tests, but nothing blocking that you can handle before merging 🎉

jandrade commented on packages/wonder-blocks-form/src/components/text-field.md:
```diff
@@ -1,15 +1,91 @@
+    constructor(props) {
+        super(props);
+        this.state = {
+            value: "",
```
One more thing, could you please add a default value here? so the story shows how the input would look with a number on it.

jandrade commented on packages/wonder-blocks-form/src/components/text-field.md:
```diff
@@ -1,15 +1,91 @@
+    constructor(props) {
+        super(props);
+        this.state = {
+            value: "",
```
Same with this story.

https://www.chromatic.com/pullrequest?activeElementId=comparison-test-key-60ae86565fbe3da84802e7cd-1200&appId=5e1bf4b385e3fb0020b7073c&number=1126&view=changes

RivasCVA commented:
What changed:
* Added better unit testing to test the core HTML instead of React using the `toContainMatchingElement` matcher
* Added a default `value` for the `number` and `password` examples in Storybook (and Styleguidist) so it shows by default how the TextField would look with a number or a password.
* Removed the use of `() => void 0` and replaced it with `() => {}` instead.

jandrade reviewed (approved):
LGTM!, thanks for the changes 🚢

RivasCVA commented:
The parent pull-request ([#1124]](https://github.com/Khan/wonder-blocks/pull/1124)) has been merged into `master`, but this branch (`wb-1049`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.
